lib/model/message_list.dart

```diff
@@ -420,6 +420,35 @@ class MessageListView with ChangeNotifier, _MessageSequence {
     notifyListeners();
   }
 
+  void maybeUpdateMessageFlags(UpdateMessageFlagsEvent event) {
+    final isAdd = switch (event) {
+      UpdateMessageFlagsAddEvent()    => true,
+      UpdateMessageFlagsRemoveEvent() => false,
+    };
+
+    bool didUpdateAny = false;
+    if (event.all) {
+      for (final message in messages) {
+        (isAdd ? message.flags.add : message.flags.remove).call(event.flag);
+        didUpdateAny |= true;
```

Comment:
nit: this can be simplified to a plain assignment, which is equivalent:
```suggestion
        didUpdateAny = true;
```